src/OrchardCore.Modules/OrchardCore.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -119,8 +119,7 @@ public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpd
                     if (possibleConflicts.Any())
                     {
                         var hasConflict = false;
-                        if (possibleConflicts.Any(x => x.ContentItemId != model.ContentItem.ContentItemId) ||
-                            possibleConflicts.Any(x => !string.IsNullOrEmpty(x.ContainedContentItemId) && x.ContainedContentItemId != model.ContentItem.ContentItemId))
+                        if (possibleConflicts.Any(x => x.ContentItemId != model.ContentItem.ContentItemId && (string.IsNullOrEmpty(x.ContainedContentItemId) || x.ContainedContentItemId != model.ContentItem.ContentItemId)))
```

Comment:
Hmm, I don't disagree with the potential issue, but this is not the correct fix (or if it works for this, it is breaking something else)

It definitely needs to evaluate the list of possible conflicts twice.

I thin you maybe confused as to what model.ContentItem.ContentItemId id is here. It will always be the Term, when evaluating the Term routing.

It may well be that the other part of your fix to maintain content item ids was actually enough to fix the issue.

Reply:
I really think the original version is wrong: When model.ContentItem represents a Term, `possibleConflicts.Any(x => x.ContentItemId != model.ContentItem.ContentItemId)` will be _always_ be true, making the whole expression true. So, we have a “conflict” even if the index entry represents the exact same Term (witch is not a “conflict”!). That can be verified, with the "replicate steps" I shown above.

After giving this some more thought, I still believe the solution works in all cases. But noticed that the `string.IsNullOrEmpty(x.ContainedContentItemId)` bit is not doing anything useful:

We just need to find if the index entry does not represent our model.ContentItem (in that case another content item is using the same Path, and we have a conflict). Assuming that every ContentItemId is unique (even a Taxonomy and a Term can't have the same ContentItemId), we just need to do:

```
if (possibleConflicts.Any(x => x.ContentItemId != model.ContentItem.ContentItemId && x.ContainedContentItemId != model.ContentItem.ContentItemId)))
{
	// An index entry represents a different item. We have a conflict. Add model error.
	[...]
} 
```

Does this make sense? Or do I need another cup of coffee? :-)

I made the change, so it can be tested.

(There is yet another problem: we are not running this test (looking for conflicts) when settings.ManageContainedItemRoutes is true and model.Absolute is false, witch allows for duplicate paths like "categories/travel". But I'm not touching that now.)